x/wasm/types/tx.go

```diff
@@ -2,12 +2,43 @@ package types
 
 import (
 	"encoding/json"
+	"errors"
 	"strings"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 )
 
+// RawContractMessage defines a json message that is sent or returned by a wasm contract.
+type RawContractMessage []byte
+
+func (r RawContractMessage) MarshalJSON() ([]byte, error) {
+	// copied from json.RawMessage#MarshalJSON
+	if r == nil {
+		return []byte("null"), nil
+	}
+	return r, nil
+}
+
+func (a *RawContractMessage) UnmarshalJSON(b []byte) error {
+	// copied from json.RawMessage#UnmarshalJSON
+	if a == nil {
+		return errors.New("unmarshalJSON on nil pointer")
+	}
+	*a = append((*a)[0:0], b...)
```

Comment:
Using this type rather than json.RawMessage everywhere gives flexibility.

What about doing `ValidateBasic()` in `UnmarshalJSON()`? Then we guarantee it is always valid